views/UserManagementView.vue

```diff
@@ -0,0 +1,105 @@
+<script setup lang="ts">
+import { z } from 'zod'
+import type { FormError, FormSubmitEvent } from '#ui/types'
+import { passwordMinimalLength } from '~/utilities/Utils'
+import { getAllUsernamesFromFirebase } from "~/model/FirebaseModel";
+
+// Props
+const props = defineProps({
+  username: {
+    type : String,
+    required: true
+  },
+})
+
+// Emits
+const emit = defineEmits(['change-info-event', 'delete-account-event'])
+
+// Constants
+const schema = z.object({
+  newUsername: z.string().optional(),
+  newEmail: z.string().email('Invalid email').optional(),
+  newPassword: z.string().min(passwordMinimalLength, 'Must be at least ' + passwordMinimalLength + ' characters.').optional(),
+  password: z.string()
+})
+const user = useCurrentUser()!
+let usernames: string[] = await getAllUsernamesFromFirebase()
```

Comment:
Doing this is dangerous compared to the former way of doing it, if a user a goes in the UserManagementView and waits a bit of it is possible that a new account is created by user b before the user a updates theirs. Usernames won't be updated and they could technically have the same username.